sphinx/environment/__init__.py

```diff
@@ -214,6 +211,14 @@ def __init__(self, srcdir, doctreedir, config):
         # attributes of "any" cross references
         self.ref_context = {}
 
+        self.init_managers()
+
+    def init_managers(self):
```

Comment:
@willingc  Thank you for comment.

But the main concern of the modules is not event handling. The environment instance delegates a part of its job to these modules.

The IndexEntries module does following operations:
* collects the index information from docs
* allows to refer them
* builds index pages for some buildes.

The Toctree module does following:
* collects ToC-tree information from docs
* assigns section and figure numbers using toctree info
* builds toctree for some builders
* builds sidebar for some builders (mainly HTML builders)

Thanks,